### PR TITLE
[APP] Android RPC

### PR DESCRIPTION
**Pull request by javelinjs**
- [x] Build android JNI automatically
- [x] OpenCL support & test
- [x] Build & sign .apk
- [x] Generate android .o & .so in python (seems that we can simply re-use `tvm.contrib.cc.create_shared`)
- [x] reconnect when rpc session ends
- [x] android rpc UT & Instrumentation test (integration test by `apps/android_rpc/tests/android_rpc_test.py`)

connected
error

Tested on my Redmi 4X Android Phone:
Processor: AArch64 Processor rev 4 (aarch64)
GLES: Qualcomm, Adreno (TM) 505, OpenGL ES 3.1 V@145.0 (GIT@Iebe23be877)
```
$ python3 android_rpc_test.py 
TVM: Initializing cython mode...
[00:38:20] src/codegen/llvm/codegen_llvm.cc:75: set native vector to be 32 for target aarch64
[00:38:20] src/runtime/opencl/opencl_device_api.cc:194: Initialize OpenCL platform 'Apple'
[00:38:20] src/runtime/opencl/opencl_device_api.cc:214: opencl(0)='Iris' cl_device_id=0x1024500
[00:38:21] src/codegen/llvm/codegen_llvm.cc:75: set native vector to be 32 for target aarch64
Run GPU test ...
0.000150427 secs/op
Run CPU test ...
0.000331302 secs/op
```

### CODE REVIEW

File: apps/android_rpc/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<resources>
+    <string name="app_name">TVM Rpc</string>
```
Comment (tqchen) [inline apps/android_rpc/app/src/main/res/values/strings.xml]: Rpc->RPC

File: python/tvm/contrib/cc.py [hunk 2]
```diff
@@ -7,7 +7,8 @@
 def create_shared(output,
```
Comment (tqchen) [inline python/tvm/contrib/cc.py]: Maybe it is easier to create a ndk.py to do the actual cross compile to android? We can bundle quite a lot of things into it, like default compiler search.

Comment (javelinjs) [inline python/tvm/contrib/cc.py]: Yes, I created one. forgot to commit.

Comment (tqchen) [inline python/tvm/contrib/cc.py]: since there is not too much reuse, maybe it is simpler to put the logics in ndk.py and not call create_shared here

**Comment by javelinjs** [comment]
@tqchen Ready to merge.
I'll provide documents in another PR (and java runtime documents)

File: python/tvm/contrib/ndk.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    Parameters
+    ----------
+    output : str
+        The target shared library.
+
+    objects : list
+        List of object files.
+
+    options : str
```
Comment (tqchen) [inline python/tvm/contrib/ndk.py]: List of str,  optional

**Pull request merged**